### PR TITLE
Add CBMC proof of processDataBlock

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of processDataBlock function

Description
-----------
This PR adds files required for the CBMC proof of processDataBlock function. 

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/processDataBlock/processDataBlock_harness.c [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                                  uint32_t uBlockSize,
+                                  OtaPalStatus_t * pCloseResult,
+                                  uint8_t * pPayload );
+
+/* Stub to validate incoming data block. */
+bool validateDataBlock( const OtaFileContext_t * pFileContext,
+                        uint32_t blockIndex,
+                        uint32_t blockSize )
+{
+    bool status;
```
Comment (gshvang) [inline test/cbmc/proofs/processDataBlock/processDataBlock_harness.c:44]: Assumption os status values  is missing.

File: test/cbmc/proofs/processDataBlock/processDataBlock_harness.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    uint32_t uBlockIndex;
+    uint32_t uBlockSize;
+    OtaPalStatus_t closeResult;
+    uint8_t * pPayload;
+
+    uint32_t fileBitmapSize;
+
+    /* The maximum number of Blocks is defined by the OTA_MAX_BITMAP_SIZE and the size of
+     * the receiver block bitmap cannot exceed that. */
+    fileContext.pRxBlockBitmap = ( uint8_t * ) malloc( OTA_MAX_BLOCK_BITMAP_SIZE );
```
Comment (gshvang) [inline test/cbmc/proofs/processDataBlock/processDataBlock_harness.c:67]: Freeing this memory is missing.

File: test/cbmc/source/stubs.c [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+                              const char * const pTimerName,
+                              const uint32_t timeout,
+                              OtaTimerCallback_t callback )
+{
+    OtaOsStatus_t status;
+
+    /* status must have values only from the OtaOsStatus_t enum. */
+    __CPROVER_assume( ( status >= OtaOsSuccess ) && ( status <= OtaOsTimerDeleteFailed ) );
+
+    __CPROVER_assert( ( otaTimerId != OtaSelfTestTimer ) || ( otaTimerId != OtaRequestTimer ),
```
Comment (AniruddhaKanhere) [inline test/cbmc/source/stubs.c:54]: Should this not be:
```suggestion
    __CPROVER_assert( ( otaTimerId == OtaSelfTestTimer ) || ( otaTimerId == OtaRequestTimer ),
```
or am I missing something?

File: test/cbmc/source/stubs.c [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+}
+
+OtaOsStatus_t stopTimerStub( OtaTimerId_t otaTimerId )
+{
+    OtaOsStatus_t status;
+
+    /* status must have values only from the OtaOsStatus_t enum. */
+    __CPROVER_assume( ( status >= OtaOsSuccess ) && ( status <= OtaOsTimerDeleteFailed ) );
+
+    __CPROVER_assert( ( otaTimerId != OtaSelfTestTimer ) || ( otaTimerId != OtaRequestTimer ),
```
Comment (AniruddhaKanhere) [inline test/cbmc/source/stubs.c:71]: Similarly here?

File: test/cbmc/source/stubs.c [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+    __CPROVER_assert( pFileContext != NULL,
+                      "Error: pFileContext in the otaAgent is statically initialized and hence cannot be NULL." );
+
+    return state;
+}
+
+void otaAppCallback( OtaJobEvent_t eEvent,
+                     const void * pData )
+{
```
Comment (AniruddhaKanhere) [inline test/cbmc/source/stubs.c:92]: Do we gurantee that on a given `eEvent` corresponding `pData` is non-NULL?

File: test/cbmc/source/stubs.c [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file stubs.c
+ * @brief Includes the definition of all the stubs required for the CBMC proofs.
+ */
+/*  Ota Agent includes. */
+#include "ota.h"
+#include "stubs.h"
```
Comment (AniruddhaKanhere) [inline test/cbmc/source/stubs.c:29]: An overarching comment: I would have liked it more if we had added stubs in one separate PR.

File: test/cbmc/source/stubs.c (hunk 5 above)
Comment (Bujain) [reply test/cbmc/source/stubs.c]: No, there is no guarantee that the corresponding `pData` is always non-NULL, it can be NULL too.

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**